guides/octane-auth-torii-with-github.md

```diff
@@ -0,0 +1,668 @@
+line of your app, after “Secret API Key:.”
+
+Of course, clicking on the “Log in to GitHub” button will not do anything,
+because we have not yet defined the `torii` authenticator or the `github`
+provider and because the template hasn’t connected the button to the `login()`
+action. Those are the next two steps.
+
+### Define Torii Authenticator and GitHub Provider
+
+First, generate an authenticator for Torii.
+
+```
+ember generate authenticator torii
+```
+
+This creates a file, `app/authenticators/torii.js`, the contents of which we
+can replace with:
+
+```javascript
+// app/authenticators/torii.js
+import { inject as service } from "@ember/service";
+import ToriiAuthenticator from "ember-simple-auth/authenticators/torii";
+
+export default ToriiAuthenticator.extend({
+  torii: service()
+});
+```
+
+Next, we create the GitHub provider by creating a new folder,
+`app/torii-providers/`, and creating a file inside it,
+`app/torii-providers/github.js`:
+
+```javascript
+// app/torii-providers/github.js
+import GitHubOAuth2Provider from "torii/providers/github-oauth2";
+
+export default GitHubOAuth2Provider.extend({
+});
+```
+
+### Enable Action on `login` Template Button and Authenticate
+
+Change the `login` template to send the `login()` action when the button is pressed.
+
+```handlebars
+{{!-- app/templates/login.hbs --}}
+<h3>Login Route</h3>
+<button onclick={{this.login}}>Log in to GitHub</button>
+```
+
+The “Log Out” button was already associated with the `logout()` action, but
+because it has been impossible to authenticate, it has also been impossible to
+see that button.
+
+If you start up the server again and now click on “Log In to GitHub,” a small
+popup window should appear asking for authorization, showing a GitHub icon and
+your GitHub user avatar. Upon clicking the green “Authorize” button, the popup
+should disappear and dump you back to your app, but now on the `index` route.
+Underneath “Index Route,” you should also now see a line displaying the GitHub
+authorization code.
+
+We’ve now established the mechanism to obtain an authorization code from
+GitHub. This doesn't authorize us fully to use the GitHub APIs, although
+`ember-simple-auth` considers us authenticated at this point, hence
+`this.session.isAuthenticated` evaluates to `true`.
+
+The “Logout” button in the `application` template now appears, and if you
+click on it, you will be dumped back to the `login` route. Logging in,
+however, no longer requires authorization via the GitHub popup, because you
+have already authorized your app. The popup still appears, but it flashes for
+a second and then disappears.
+
+This completes the second step of the outline at the top of this guide. Next
+is performing the token exchange to enable getting user data from GitHub.
+
+## Deploy a Gatekeeper Server for Token Exchange and Connect It to Ember
+
+Here the app has to move out of Emberdom for a bit, because GitHub will not
```

Comment:
I'm wondering what is meant by `Emberdom` - can you outline it?

Reply:
I've clarified the wording. I just meant that the next step happens outside of the Ember app. This was, conceptually, the hardest part for me to understand.